src/main/java/io/fusionauth/http/security/SecurityTools.java

```diff
@@ -75,25 +76,45 @@ public static SSLContext clientContext(Certificate certificate) throws GeneralSe
   }
 
   public static Certificate parseCertificate(String certificate) throws CertificateException {
+    return parseCertificates(certificate)[0];
+  }
+
+  public static Certificate[] parseCertificates(String certificates) throws CertificateException {
     CertificateFactory factory = CertificateFactory.getInstance("X.509");
-    byte[] certBytes = parseDERFromPEM(certificate, CERT_START, CERT_END);
-    return factory.generateCertificate(new ByteArrayInputStream(certBytes));
+    byte[] certBytes = parseDERFromPEM(certificates, CERT_START, CERT_END);
+    return factory.generateCertificates(new ByteArrayInputStream(certBytes)).toArray(new Certificate[]{});
   }
 
+  /**
+   * Parses all objects in a PEM-formatted string into a byte[].
+   */
   public static byte[] parseDERFromPEM(String pem, String beginDelimiter, String endDelimiter) {
-    int startIndex = pem.indexOf(beginDelimiter);
-    if (startIndex < 0) {
-      throw new IllegalArgumentException("Invalid PEM format");
-    }
+    // Allocate an initial buffer to hold 2 certificates.
+    ByteArrayOutputStream outBytes = new ByteArrayOutputStream(4000);
 
-    int endIndex = pem.indexOf(endDelimiter);
-    if (endIndex < 0) {
+    try {
+      while (!pem.trim().isEmpty()) {
+        int startIndex = pem.indexOf(beginDelimiter);
+        if (startIndex < 0) {
+          throw new IllegalArgumentException("Invalid PEM format");
+        }
+
+        int endIndex = pem.indexOf(endDelimiter);
+        if (endIndex < 0) {
+          throw new IllegalArgumentException("Invalid PEM format");
+        }
+
+        String base64 = pem.substring(startIndex + beginDelimiter.length(), endIndex);
+
+        // Decode current chunk using the MIME decoder to strip whitespace, then skip past chunk.
+        outBytes.write(Base64.getMimeDecoder().decode(base64));
+        pem = pem.substring(endIndex + endDelimiter.length() + 5);
```

Comment:
Is this `+ 5` because the start delimiter is `BEGIN CERTIFICATE----` instead of `----BEGIN CERTIFICATE----?

Reply:
It's skipping the trailing `----` after `END CERTIFICATE`. I'll clarify that in the comment.